src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -247,7 +246,11 @@ public SocketError AcceptAsync(byte[] socketAddress, int socketAddressLen, Actio
             SocketError errorCode;
             if (TryCompleteAccept(_fileDescriptor, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
             {
-                ThreadPool.QueueUserWorkItem(_ => callback(acceptedFd, socketAddress, socketAddressLen, errorCode), null);
+                ThreadPool.QueueUserWorkItem(args =>
+                {
+                    var tup = (Tuple<int, byte[], int, SocketError>)args;
+                    callback(tup.Item1, tup.Item2, tup.Item3, tup.Item4);
```

Comment:
```callback``` here should be passed in the Tuple as well.  As it is, both the closure and the delegate are still getting allocated to pass callback in.